### PR TITLE
Update to 4.39.0

### PR DESCRIPTION
**Pull request by albilaga**

### CODE REVIEW

File: Xamarin.RevenueCat.iOS/ApiDefinitions.cs [hunk 1]
```diff
@@ -656,900 +719,989 @@ interface RCPromotionalOfferSignedData
...
+
+        // @required -(void)restorePurchasesWithCompletionHandler:(void (^ _Nonnull)(RCCustomerInfo * _Nullable, NSError * _Nullable))completionHandler __attribute__((availability(watchos, introduced=6.2))) __attribute__((availability(tvos, introduced=13.0))) __attribute__((availability(macos, introduced=10.15))) __attribute__((availability(ios, introduced=13.0)));
+        [Abstract]
+        [Export("restorePurchasesWithCompletionHandler:")]
+        void RestorePurchasesWithCompletionHandler(Action<RCCustomerInfo, NSError> completionHandler);
+
+        // @required -(void)syncPurchasesWithCompletion:(void (^ _Nullable)(RCCustomerInfo * _Nullable, NSError * _Nullable))completion;
+        [Abstract]
+        [Export("syncPurchasesWithCompletion:")]
+        void SyncPurchasesWithCompletion([NullAllowed] Action<RCCustomerInfo, NSError> completion);
```
Comment (thisisthekap) [inline Xamarin.RevenueCat.iOS/ApiDefinitions.cs:857]: rename

File: Xamarin.RevenueCat.iOS/StructsAndEnums.cs [hunk 2]
```diff
@@ -12,7 +12,7 @@ public enum RCAttributionNetwork : long
         Tenjin = 4,
         Facebook = 5,
         MParticle = 6,
-        AdServices = 7,
+        AdServices = 7
```
Comment (thisisthekap) [inline Xamarin.RevenueCat.iOS/StructsAndEnums.cs:15]: add the "," signs for all last enum entries

File: Xamarin.RevenueCatUI.iOS.Extensions/Paywall.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+using UIKit;
+
+namespace Xamarin.RevenueCatUI.iOS.Extensions
+{
+    public static class Paywall
+    {
+        public static Task<RCCustomerInfo> PresentPaywallAsync(this UIViewController topVc, RCOffering offering,
+            CancellationToken cancellationToken = default)
+        {
+            var tcs = new TaskCompletionSource<RCCustomerInfo>();
```
Comment (thisisthekap) [inline Xamarin.RevenueCatUI.iOS.Extensions/Paywall.cs:14]: cancel tcs right away if the cancellationtoken is triggered

File: Xamarin.RevenueCatUI.iOS.Extensions/Xamarin.RevenueCatUI.iOS.Extensions.csproj [hunk 4]
```diff
@@ -0,0 +1,73 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    <PropertyGroup>
+        <IsPackable>true</IsPackable>
+        <PackageId>Xamarin.RevenueCatUI.iOS.Extensions</PackageId>
+        <Version>4.39.0.1</Version>
+        <Description>Contains bindings for https://docs.revenuecat.com/docs/ios</Description>
+        <Authors>Christian Kapplmüller, Albilaga</Authors>
```
Comment (thisisthekap) [inline Xamarin.RevenueCatUI.iOS.Extensions/Xamarin.RevenueCatUI.iOS.Extensions.csproj:8]: do you want to add your full name?

File: Xamarin.RevenueCatUI.iOS/Structs.cs [hunk 5]
```diff
@@ -0,0 +1,5 @@
+﻿using System;
+
+namespace Xamarin.RevenueCatUI.iOS
```
Comment (thisisthekap) [inline Xamarin.RevenueCatUI.iOS/Structs.cs:3]: I think this file can be removed.

File: Xamarin.RevenueCatUI.iOS/clear.sh [hunk 6]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+rm -rf Output
+rm -rf Carthage
```
Comment (thisisthekap) [inline Xamarin.RevenueCatUI.iOS/clear.sh:3]: i am confused. are we using Carthage here?

**Review by thisisthekap — CHANGES_REQUESTED** [review]

File: Xamarin.RevenueCatUI.iOS/clear.sh (hunk 6 above)
Comment (albilaga) [reply Xamarin.RevenueCatUI.iOS/clear.sh]: yeah we are not using carthage. removed

**Review by thisisthekap — APPROVED** [review]

**Pull request merged**